gcloud-java-dns/src/main/java/com/google/gcloud/dns/testing/LocalDnsHelper.java

```diff
@@ -0,0 +1,1360 @@
+      this.domain = domain;
+      this.reason = reason;
+      this.status = status;
+    }
+
+    Response response(String message) {
+      try {
+        return new Response(code, toJson(message));
+      } catch (IOException e) {
+        return Error.INTERNAL_ERROR.response("Error when generating JSON error response.");
+      }
+    }
+
+    private String toJson(String message) throws IOException {
+      Map<String, Object> errors = new HashMap<>();
+      errors.put("domain", domain);
+      errors.put("message", message);
+      errors.put("reason", reason);
+      Map<String, Object> args = new HashMap<>();
+      args.put("errors", ImmutableList.of(errors));
+      args.put("code", code);
+      args.put("message", message);
+      args.put("status", status);
+      return jsonFactory.toString(ImmutableMap.of("error", args));
+    }
+  }
+
+  private class RequestHandler implements HttpHandler {
+
+    /**
+     * Chooses the proper handler for a request.
+     */
+    private Response pickHandler(HttpExchange exchange, CallRegex regex) {
+      String path = BASE_CONTEXT.relativize(exchange.getRequestURI()).getPath();
+      String[] tokens = path.split("/");
+      String projectId = tokens != null && tokens.length > 0 ? tokens[0] : null;
+      String zoneName = tokens != null && tokens.length > 2 ? tokens[2] : null;
+      String changeId = tokens != null && tokens.length > 4 ? tokens[4] : null;
+      String query = BASE_CONTEXT.relativize(exchange.getRequestURI()).getQuery();
+      switch (regex) {
+        case CHANGE_GET:
+          return handleChangeGet(projectId, zoneName, changeId, query);
+        case CHANGE_LIST:
+          return handleChangeList(projectId, zoneName, query);
+        case ZONE_GET:
+          return handleZoneGet(projectId, zoneName, query);
+        case ZONE_DELETE:
+          return handleZoneDelete(projectId, zoneName);
+        case ZONE_LIST:
+          return handleZoneList(projectId, query);
+        case PROJECT_GET:
+          return handleProjectGet(projectId, query);
+        case RECORD_LIST:
+          return handleDnsRecordList(projectId, zoneName, query);
+        case ZONE_CREATE:
+          try {
+            return handleZoneCreate(exchange, projectId, query);
+          } catch (IOException ex) {
+            return Error.BAD_REQUEST.response(ex.getMessage());
+          }
+        case CHANGE_CREATE:
+          try {
+            return handleChangeCreate(exchange, projectId, zoneName, query);
+          } catch (IOException ex) {
+            return Error.BAD_REQUEST.response(ex.getMessage());
+          }
+        default:
+          return Error.INTERNAL_ERROR.response("Operation without a handler.");
+      }
+    }
+
+    @Override
+    public void handle(HttpExchange exchange) throws IOException {
+      String requestMethod = exchange.getRequestMethod();
+      String rawPath = exchange.getRequestURI().getRawPath();
+      for (CallRegex regex : CallRegex.values()) {
```

Comment:
We can either do a loop which tests match to the regular expression and gives us the enum for switch, or we can do bunch of if-elseif statements. I chose this because even though it may be "looping twice" through the options (a constant number), it seems more readable and easier to maintain compare to 10 if-elseif-else statements.